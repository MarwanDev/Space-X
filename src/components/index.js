export { default as Navbar } from './Navbar/Navbar';
export { default as Rockets } from './Rockets/Rockets';
export { default as Missions } from './Missions/Missions';
export { default as MyProfile } from './MyProfile/MyProfile';

