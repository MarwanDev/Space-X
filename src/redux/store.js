import { configureStore } from "@reduxjs/toolkit";
import rocketReducer from "./RocketSlice/rocketsSlice";


const store = configureStore({
  reducer: {
    rockets: rocketReducer,
  },

});

export default store;
