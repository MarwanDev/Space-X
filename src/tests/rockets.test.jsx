import { expect, it, describe } from "vitest";
import renderer from "react-test-renderer";
import store from "../redux/store";
import reducer, {
  reserveRocket,
  cancelReservation,
} from "../redux/RocketSlice/rocketsSlice";
import { Provider } from "react-redux";
import { Rockets } from "../components";
import { render } from "@testing-library/react";

describe("Rockets", () => {
  it("should reserve rockets correctly", () => {
    const state = {
      rockets: [
        {
          id: "0",
          name: "Rocket 1",
          description: "Rocket 1",
          image: "image",
        },
        {
          id: "1",
          name: "Rocket 2",
          description: "Rocket 2",
          image: "image",
        },
        {
          id: "2",
          name: "Rocket 3",
          description: "Rocket 3",
          image: "image",
        },
      ],
      isLoading: false,
    };
    expect(reducer(state, reserveRocket("1"))).toEqual({
      rockets: [
        {
          id: "0",
          name: "Rocket 1",
          description: "Rocket 1",
          image: "image",
        },
        {
          id: "1",
          name: "Rocket 2",
          description: "Rocket 2",
          image: "image",
          reserved: true,
        },
        {
          id: "2",
          name: "Rocket 3",
          description: "Rocket 3",
          image: "image",
        },
      ],
      isLoading: false,
    });
  });

  it("should cancel rockets reservation correctly", () => {
    const state = {
      rockets: [
        {
          id: "0",
          name: "Rocket 1",
          description: "Rocket 1",
          image: "image",
          reserved: true,
        },
        {
          id: "1",
          name: "Rocket 2",
          description: "Rocket 2",
          image: "image",
          reserved: true,
        },
        {
          id: "2",
          name: "Rocket 3",
          description: "Rocket 3",
          image: "image",
          reserved: true,
        },
      ],
      isLoading: false,
    };
    expect(reducer(state, cancelReservation("1"))).toEqual({
      rockets: [
        {
          id: "0",
          name: "Rocket 1",
          description: "Rocket 1",
          image: "image",
          reserved: true,
        },
        {
          id: "1",
          name: "Rocket 2",
          description: "Rocket 2",
          image: "image",
          reserved: false,
        },
        {
          id: "2",
          name: "Rocket 3",
          description: "Rocket 3",
          image: "image",
          reserved: true,
        },
      ],
      isLoading: false,
    });
  });
  
  it("renders rocket correctly", () => {
    const rockets = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>
      )
      .toJSON();
    expect(rockets).toMatchSnapshot();
  });
  
  it("renders rocket correctly", () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>
    );
  });
});


