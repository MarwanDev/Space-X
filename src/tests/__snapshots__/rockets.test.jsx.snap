// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Rockets > renders rocket correctly 1`] = `
<div>
  <p>
    Bear With Us, Your Rockets are Loading
  </p>
</div>
`;

exports[`renders rocket correctly 1`] = `
<div>
  <p>
    Bear With Us, Your Rockets are Loading
  </p>
</div>
`;
